src/Build/Evaluation/LazyItemEvaluator.LazyItemOperation.cs

```diff
@@ -110,14 +110,14 @@ protected void DecorateItemsWithMetadata(ICollection<I> items, ImmutableList<Par
 
                     // Do not expand properties as they have been already expanded by the lazy evaluator upon item operation construction.
                     // Prior to lazy evaluation ExpanderOptions.ExpandAll was used.
-                    const ExpanderOptions metadataExpansionOptions = ExpanderOptions.ExpandMetadata | ExpanderOptions.ExpandItems;
```

Comment:
I'm wondering about my comment - the property function `$([System.String]::new('%(RelativeDir)').TrimEnd('\'))` would have been expanded at operation construction time and that should have resulted in `%(RelativeDir)`, is that correct? If so, then shouldn't that have been expanded at `operation application` time because we have ...

`const ExpanderOptions metadataExpansionOptions = ExpanderOptions.ExpandMetadata | ExpanderOptions.ExpandItems;`

... like you pointed out? I am asking out of curiosity, but didn't actually go and confirm this with a simple printf!